flybywiresim/aircraft
# feat(A380X): A380X support for Lighting Presets

## Summary of Changes

- A32NX/A380X: 
	- Improved shared code and separation for aircraft 
	- Smooth loading has been improved to accommodate for different fps - takes a configurable amount of time (LVAR:LIGHTING_PRESET_LOAD_TIME) to move potentiometers from 0 to 100% (within limits ~1-10sec). 
	- Further fixes and improvements also for the A32NX
	- OBS: Changes a few value ranges for the presets ini file - users will have to redo/fix these. Values are not all normalized to 0..100 independent of the value the actual aircraft code uses. 

- A380X:
	- Support for Lighting Presets 
	- Added LIGHT POTENTIOMETER 80, 81 and 82 to A380X RMPs

Discord username (if different from GitHub): cdr_maverick

## Testing instructions
A32NX:
Test if flyPad lighting presets work as expected (regression testing)

A380X:
Test if flyPad lighting presets work as expected with all lights available in the A380X
- use safe and load of presets
- use manual editing of the lighting preset file for lights where cockpit controls do not yet work (RMPs)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
